check-test-suite.py

```diff
@@ -0,0 +1,125 @@
+# of CI can be skipped
+ignore_prefixes = ['.github', '.idea', '.asf.yaml', '.backportrc.json', '.codecov.yml', '.dockerignore', '.gitignore',
+                   '.lgtm.yml', 'CONTRIBUTING.md', 'setup-hooks.sh', 'upload.sh', 'dev', 'distribution/docker',
+                   'distribution/asf-release-process-guide.md', '.travis.yml', 'check-test-suite.py']
+# these files are docs changes
+# if changes are limited to this set then we can skip web-console and java
+# if no changes in this set we can skip docs
+docs_prefixes = ['docs/', 'website/']
+# travis docs job name
+docs_jobs = ['docs']
+# these files are web-console changes
+# if changes are limited to this set then we can skip docs and java
+# if no changes in this set we can skip web-console
+web_console_prefixes = ['web-console/']
+# travis web-console job name
+web_console_jobs = ['web console', 'web console end-to-end test']
+
+
+def check_ignore(file):
+    is_always_ignore = True in (file.startswith(prefix) for prefix in ignore_prefixes)
+    if is_always_ignore:
+        print("found ignorable file change: {}".format(file))
+    return is_always_ignore
+
+def check_docs(file):
+    is_docs = True in (file.startswith(prefix) for prefix in docs_prefixes)
+    if is_docs:
+        print("found docs file change: {}".format(file))
+    return is_docs
+
+def check_console(file):
+    is_console = True in (file.startswith(prefix) for prefix in web_console_prefixes)
+    if is_console:
+        print("found web-console file change: {}".format(file))
+    return is_console
+
+def check_should_run_suite(suite, diff_files):
+    if suite in always_run_jobs:
+        # you gotta do what you gotta do
+        return True
+
+    all_ignore = True
+    any_docs = False
+    all_docs = True
+    any_console = False
+    all_console = True
+
+    for f in diff_files:
+        all_ignore = all_ignore and check_ignore(f)
+        is_docs = check_docs(f)
+        any_docs = any_docs or is_docs
+        all_docs = all_docs and is_docs
+        is_console = check_console(f)
+        any_console = any_console or is_console
+        all_console = any_console and is_console
+
+    if all_ignore:
+        return False
+    if suite in docs_jobs:
+        return any_docs
+    if all_docs:
+        return False
+    if suite in web_console_jobs:
+        return any_console
+    if all_console:
+        return False
+
+    return True
+
+def failWithUsage():
+    sys.stderr.write("usage: check-test-suite.py <test-suite-name>\n")
+    sys.stderr.write("  e.g., check-test-suite.py docs")
+    sys.exit(1)
+
+suite_name = ""
+
+if len(sys.argv) == 1:
+    if 'TRAVIS_JOB_NAME' in os.environ:
+        suite_name = os.environ['TRAVIS_JOB_NAME']
+    else:
+        failWithUsage()
+elif len(sys.argv) == 2:
+    suite_name = sys.argv[1]
+else:
+    failWithUsage()
+
+
+all_changed_files_string = subprocess.check_output("git diff --name-only HEAD~1", shell=True).decode('UTF-8')
```

Comment:
What happens if there's a PR with multiple commits? I think we need a different git command to find all the commits that are new compared to the base branch

Reply:
Also, can you comment on how this PR will ensure all tests are still run on master or release branches like `0.21.1`

Reply:
I think per discussion in https://github.com/DefinitelyTyped/DefinitelyTyped/issues/1314 which i came across when searching the interwebs for how to best do this stuff, that it would appear this command works specifically for travis PR builds because of the way it runs them on top of the branch you are merging into.

You're right that this wouldn't be sufficient for branch builds, so this script should either always indicate that non-PR builds should run the entire suite, or, have a different command to use in that case. The way the current diff check is performed could also probably work I think, https://github.com/apache/druid/blob/master/.travis.yml#L176 if we want to enable selectively running tests on branch commits.

Thinking about it though, it might be safest for now to just always run the entire set on branch builds, at least for now, in case this script messes something up and skips too much